src/components/BaseModal/BaseModal.jsx

```diff
@@ -0,0 +1,41 @@
+import { ModalBackground, ModalContainer, ModalContainerCloseButWrapper, CloseButImg } from "./BaseModal.styled";
+import closeCross from "./assets/closeCross.svg"
+import { useEffect, useState } from "react";
+
+export const BaseModal = ({ifShown = true}) => {
+
+    const [ stateIfShown, setStateIfShown ] = useState(ifShown);
+
+    const keyboardListener = (e) => {
+        if (e.key === "Escape") {
+            closeModalHandler();
+        }
+    }
+
+    function closeModalHandler() {
+        setStateIfShown(false);
+        window.onscroll = () => window.scrollTo();
+        document.removeEventListener("keydown", keyboardListener);
+    }
+
+    useEffect(() => {
+        window.onscroll = () => window.scrollTo(0, 0);
```

Comment:
I am not sure it's correct to move the page scrolling up when a user opens the modal for example on the end of the page.
You should just disable scroll while modal is opened, not to change its position